File: src/IntentSystem.Projection/Models/ImplementationIssuePacket.cs
namespace IntentSystem.Projection.Models;

public sealed record ImplementationIssuePacket
{
    public required string IssueTitle { get; init; }

    public required IssueKind IssueKind { get; init; }

    public required string SourceExecutionUnit { get; init; }

    public required string Goal { get; init; }

    public required IReadOnlyList<string> InScope { get; init; }

    public required IReadOnlyList<string> OutOfScope { get; init; }

    public required string TargetRepo { get; init; }

    public required string TargetPath { get; init; }

    public required string TargetPart { get; init; }

    public required IReadOnlyList<string> Dependencies { get; init; }

    public required IReadOnlyList<string> IntentReferences { get; init; }

    public required IReadOnlyList<string> RulesAndSpecs { get; init; }

    public required IReadOnlyList<string> AcceptanceCriteria { get; init; }

    public required IReadOnlyList<string> VerificationEvidence { get; init; }

    public required string ReviewMode { get; init; }

    public required string CompletionAction { get; init; }

    public required string LandingPolicy { get; init; }
}

File: src/schema/implementation-issue-packet.ts
import { z } from 'zod'

import { IssueKindSchema } from './issue-kind.js'

const stringListSchema = z.array(z.string())

export const ImplementationIssuePacketSchema = z.object({
  issue_title: z.string().min(1),
  issue_kind: IssueKindSchema,
  source_execution_unit: z.string().min(1),
  goal: z.string().min(1),
  target_repo: z.string().min(1),
  target_part: z.string().min(1),
  target_path: z.string().min(1),
  review_mode: z.string().min(1),
  completion_action: z.string().min(1),
  landing_policy: z.string().min(1),
  parent_intent_root: z.string().min(1),
  dependencies: stringListSchema,
  intent_references: stringListSchema,
  rules_and_specs: stringListSchema,
  acceptance_criteria: stringListSchema.min(1),
  in_scope: stringListSchema.min(1),
  out_of_scope: stringListSchema,
})

export type ImplementationIssuePacket = z.infer<typeof ImplementationIssuePacketSchema>
